### PR TITLE
Automate the catalog refresh by job scheduling package in api

### PR DESCRIPTION
**Pull request by pratap0007**
This patch creates a cron job with help job scheduling
package(https://github.com/go-co-op/gocron) in api and does
catalog refresh on an interval and default interval is 30 minutes

Signed-off-by: Shiv Verma <shverma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [ ] Run UI Unit Tests, Lint Checks with `make ui-check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/.env.dev [hunk 1]
```diff
@@ -21,6 +21,8 @@ JWT_SIGNING_KEY="TektonHub"
 ACCESS_JWT_EXPIRES_IN="1d"
 REFRESH_JWT_EXPIRES_IN="1d"
 
+CATALOG_REFRESH_INTERVAL="30m"
```
Comment (sm43) [inline api/.env.dev:24]: this will be required in api configmap and deployment right?

Comment (pratap0007) [reply api/.env.dev:24]: yes

File: api/pkg/service/catalog/syncer.go [hunk 2]
```diff
@@ -47,7 +49,7 @@ func NewSyncer(api app.BaseConfig, clonePath string) *syncer {
 	return &syncer{
 		db:        app.DBWithLogger(api.Environment(), api.DB(), logger),
 		logger:    logger.SugaredLogger,
-		limit:     make(chan bool, 1),
+		limit:     make(chan bool, 2),
```
[the remark this answers is not in the document]
Comment (pratap0007) [reply api/pkg/service/catalog/syncer.go:52]: it's my fault and should be 1

**Comment by piyush-garg** [comment]
we can remove the cron job too now which we have for catalog refresh

File: api/cmd/api/main.go [hunk 3]
```diff
@@ -94,7 +94,7 @@ func main() {
 
 	// Add resources to database
```
Comment (sm43) [inline api/cmd/api/main.go:95]: need to update comment :)

File: api/pkg/db/initializer/initializer.go [hunk 4]
```diff
@@ -253,32 +252,3 @@ func checkIfApiServerAccountExists(db *gorm.DB, logger *log.Logger) error {
 
 	return nil
 }
-
```
Comment (sm43) [inline api/pkg/db/initializer/initializer.go:256]: line no: 240
 err := q.First(&model.Account{}).Error; err == gorm.ErrRecordNotFound

File: api/pkg/service/catalog/syncer.go [hunk 5]
```diff
@@ -108,10 +111,70 @@ func (s *syncer) Run() {
...
+
+	getCatalogRefreshInterval, err := app.ComputeDuration(os.Getenv("CATALOG_REFRESH_INTERVAL"))
+	if err != nil {
+		s.logger.Error(err)
+	}
+
+	// Checks catalog refresh interval time and do catalog refresh only when it's valid
+	if getCatalogRefreshInterval.Seconds() != 0 {
+
+		fmt.Println("380084384509459845095---", getCatalogRefreshInterval.Seconds())
```
Comment (sm43) [inline api/pkg/service/catalog/syncer.go:155]:
```suggestion

```

**Pull request merged**